shielded-pool/src/component.rs

```diff
@@ -216,25 +227,34 @@ impl Component for ShieldedPool {
 
     #[instrument(name = "shielded_pool", skip(self, end_block))]
     async fn end_block(&mut self, end_block: &abci::request::EndBlock) {
-        // Set the height of the compact block, now that we got it in end_block
-        self.compact_block.height = end_block.height as u64;
+        // Get the current block height
+        let height = self
+            .state
+            .get_block_height()
+            .await
+            .expect("block height must be set");
+
+        if height != (end_block.height as u64) {
+            tracing::error!(
+                "end block height {} does not match current height {}",
+                end_block.height,
+                height
+            );
+        }
```

Comment:
I'm not sure there's much point of doing this -- at this point we're going to be badly corrupted, so adding an extra tracing message doesn't help much. Better to leave this kind of invariant checking out of the individual components IMO.

Reply:
Makes sense to me.